packages/terra-form-fieldset/src/Fieldset.jsx

```diff
@@ -61,12 +62,23 @@ const Fieldset = ({
     { 'legend-visually-hidden': isLegendHidden },
   ]);
 
+  const legendId = `terra-fieldset-legend-${uuidv4()}`;
+  const helpId = `terra-fieldset-help-${uuidv4()}`;
+
+  const addAriadescribedby = (fieldsetChildren) => {
+    const cloneChildren = [];
+    React.Children.forEach(fieldsetChildren, (child) => {
+      cloneChildren.push(React.cloneElement(child, { 'aria-labelledby': `${helpId} ${legendId}` }));
```

Comment:
```suggestion
      cloneChildren.push(React.cloneElement(child, { 'aria-labelledby': `${legendId}${helpId}` }));
```
I believe legend needs to be read before help text !!